test/adjoint/test_adjoint_mesh_seq.py

```diff
@@ -531,3 +532,86 @@ def test_enrichment_transfer(
         target = Function(mesh_seq_e.function_spaces["field"][0])
         transfer(source, target)
         self.assertAlmostEqual(norm(source), norm(target))
+
+
+class GoalOrientedBaseClass(unittest.TestCase):
+    """
+    Base class for tests with a complete :class:`GoalOrientedMeshSeq`.
+    """
```

Comment:
The proper way to set up these base classes is to have a `BaseClasses` class and then define this (and possibly others) within it's scope. See `test/test_mesh_seq.py` for an example.

Reply:
I actually followed how the existing `TrivialGoalOrientedBaseClass` was defined in this file. Should then both of these change to follow the `BaseClasses` structure?

Reply:
Ah okay I missed that - must have been an earlier implementation. That'd be great, thanks

Reply:
This actually seems like quite a bit of work now and I think it might all get restructured anyway after the introduction of `Solver` classes etc. Would you mind if I leave it as it is for now to avoid reworking it twice?

Reply:
Sure, that's fine.